eth/blockwatch/block_watcher.go

```diff
@@ -387,6 +395,11 @@ func (w *Watcher) getMissedEventsToBackfill(ctx context.Context) ([]*Event, erro
 					Number: big.NewInt(0).SetUint64(log.BlockNumber),
 					Logs:   []types.Log{},
 				}
+				// TODO: Consider extracting L1 block number directly from logs if possible
```

Comment:
I don't think this is possible right now. If you call `eth_getTransactionReceipt` on Arbitrum Rinkeby with the hash 0xe22f5ac3db57a421e6c00853c206440193fd5de5cba61c3876b88b8707b7c43e you'll see that the nested log objects in the logs array only contain the L2 block number and not the L1 block number :(

Reply:
Ok, let me remove this comment since we already know we can't address it.